docs/reference/module-api/module-overview.md

```diff
@@ -321,27 +332,59 @@ Each of these function sections takes the module, as `self`, as an argument. Thi
 | [`extendHandlers`](#extendhandlers-self) | Extend base class server-side event handlers |
 | [`queries`](#queries-self-query) | Add database query methods |
 | [`extendQueries`](#extendqueries-self) | Extend base class database query methods |
-| [`middleware`](#middleware-self) | REPLACE-ME |
-| [`tasks`](#tasks-self) | REPLACE-ME |
+| [`middleware`](#middleware-self) | Add standard Express middleware to be called on *every* request. |
+| [`tasks`](#tasks-self) | Add task functions that can be run from the command-line. |
 
 ### The extension pattern
 
 Several of these sections use an extention pattern (the sections prefixed with "extend"). These sections are used to add functionality to matching features of the module's base class. Functions in `extendMethods` enhance identically named functions in the base class' `methods` section, `extendApiRoutes` enhances API routes from the base class' `apiRoutes`, and so on.
 
 Each individual function included in the extension section takes a `_super` argument in addition to the same arguments as the original function. `_super` is the original function, which should be called within the new extension.
 
-If a piece type included the `insert` in its `extendMethods` section to alter the piece titles, it might look like this:
+If a piece type included the `generate` method in its `extendMethods` section to add a price upon generating placeholder docs, it might look like this:
 
 ```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
   extendMethods(self) {
     return {
-      insert(_super, req, piece, options) {
-        piece.title = `🆕 ${piece.title}`;
+      generate(_super, index) {
+        // Using _super with the original argument to generate a sample piece.
+        const piece = _super(index);
+        // Adding additional functionality.
+        piece.price = Math.random() * 100;
+        // Returning the generated piece, exactly as the original `generate`
+        // method does.
+        return piece;
+      }
+    };
+  }
+};
+```
+
+Or this, if extending an [async component function](#components-self):
```

Comment:
"Or this" feels a bit clunky. Maybe something like 'This example shows'. I don't feel I have an adequate understanding to fully suggest